### PR TITLE
feat: set learning rate of Gradient Boosting models

### LINKED ISSUE #168: Set learning rate of `GradientBoosting`
### Is your feature request related to a problem?

It's not possible to set the learning rate of gradient boosting.

### Desired solution

* Add a new parameter `learning_rate: float` to the initializer of `safeds.ml.classification.GradientBoosting` and `safeds.ml.regression.GradientBoosting`
* Raise a value error if `learning_rate` <= 0
* Pass it along to `learning_rate` of the wrapped `scikit-learn` model in the `fit` method

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by alex-senger**
Closes #168.

### Summary of Changes

Add `learning_rate` parameter to `GradientBoosting` classifier and regressor.

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/6.22.1) status: ✅ [SUCCESS](https://github.com/Safe-DS/Stdlib/actions/runs/4830886683)

| Descriptor  |                                 Linter                                 |Files|Fixed|Errors|Elapsed time|
|-------------|------------------------------------------------------------------------|-----|----:|------|------------|
|✅ PYTHON    |[black](https://megalinter.io/6.22.1/descriptors/python_black)          |    4|    0|     0|0.56s       |
|✅ PYTHON    |[mypy](https://megalinter.io/6.22.1/descriptors/python_mypy)            |    4|     |     0|1.67s       |
|✅ PYTHON    |[ruff](https://megalinter.io/6.22.1/descriptors/python_ruff)            |    4|    0|     0|0.03s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/6.22.1/descriptors/repository_git_diff)|yes  |     |no    |0.02s       |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/4830886683)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

File: src/safeds/ml/classical/classification/_gradient_boosting_classification.py [hunk 1]
```diff
@@ -13,12 +13,27 @@
...
-    """Gradient boosting classification."""
+    """Gradient boosting classification.
 
-    def __init__(self) -> None:
+    Parameters
+    ----------
+    learning_rate : float
+        Learning rate shrinks the contribution of each tree by `learning_rate`.
+
+
```
Comment (lars-reimann) [inline src/safeds/ml/classical/classification/_gradient_boosting_classification.py:23]: Remove line.

File: src/safeds/ml/classical/classification/_gradient_boosting_classification.py [hunk 2]
```diff
@@ -13,12 +13,27 @@
...
+    ValueError
+        If `learning_rate` is non-positive.
+    """
+
+    def __init__(self, learning_rate: float = 0.1) -> None:
         self._wrapped_classifier: sk_GradientBoostingClassifier | None = None
         self._feature_names: list[str] | None = None
         self._target_name: str | None = None
+        if learning_rate <= 0:
+            raise ValueError("learning_rate must be non-negative.")
```
Comment (lars-reimann) [inline src/safeds/ml/classical/classification/_gradient_boosting_classification.py:35]:
```suggestion
            raise ValueError("learning_rate must be positive.")
```

File: src/safeds/ml/classical/regression/_gradient_boosting_regression.py [hunk 3]
```diff
@@ -13,12 +13,27 @@
...
-    """Gradient boosting regression."""
+    """Gradient boosting regression.
 
-    def __init__(self) -> None:
+    Parameters
+    ----------
+    learning_rate : float
+        Learning rate shrinks the contribution of each tree by `learning_rate`.
+
+
```
Comment (lars-reimann) [inline src/safeds/ml/classical/regression/_gradient_boosting_regression.py:23]: Remove line.

File: src/safeds/ml/classical/regression/_gradient_boosting_regression.py [hunk 4]
```diff
@@ -13,12 +13,27 @@
...
+    ValueError
+        If `learning_rate` is non-positive.
+    """
+
+    def __init__(self, learning_rate: float = 0.1) -> None:
         self._wrapped_regressor: sk_GradientBoostingRegressor | None = None
         self._feature_names: list[str] | None = None
         self._target_name: str | None = None
+        if learning_rate <= 0:
+            raise ValueError("learning_rate must be non-negative.")
```
Comment (lars-reimann) [inline src/safeds/ml/classical/regression/_gradient_boosting_regression.py:35]:
```suggestion
            raise ValueError("learning_rate must be positive.")
```

File: tests/safeds/ml/classical/classification/test_gradient_boosting_classification.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import pytest
+from safeds.data.tabular.containers import Table
+from safeds.ml.classical.classification import GradientBoosting
+
+
+def test_should_throw_value_error_if_learning_rate_is_non_positive() -> None:
+    with pytest.raises(ValueError, match="learning_rate must be non-negative."):
```
Comment (lars-reimann) [inline tests/safeds/ml/classical/classification/test_gradient_boosting_classification.py:7]: Must be adjusted accordingly.

**Review by lars-reimann — CHANGES_REQUESTED** [review]

File: src/safeds/ml/classical/classification/_gradient_boosting_classification.py [hunk 6]
```diff
@@ -13,12 +13,27 @@
...
 
 class GradientBoosting(Classifier):
-    """Gradient boosting classification."""
+    """Gradient boosting classification.
 
-    def __init__(self) -> None:
+    Parameters
+    ----------
+    learning_rate : float
+        Learning rate shrinks the contribution of each tree by `learning_rate`.
```
[the remark this answers is not in the document]
Comment (lars-reimann) [reply src/safeds/ml/classical/classification/_gradient_boosting_classification.py]: This one's still open (same for regression).

**Review by lars-reimann — APPROVED** [review]
